### PR TITLE
builtin: create string.trim_indexes as a general function to be used in string.trim

### PR DESCRIPTION
**Pull request by l1mey112**
A refactor over `string.trim` functions to allow for getting the indexes to what parts of the string was chopped off.

This PR creates `string.trim_indexes`, similar to `string.trim` but returns the new start and end indexes without manipulating the string. The `string.trim` function is now implemented with a call to this function and passing it's return values to `string.substr` unchanged.

```v
hello_string := ' ffHello V ffff'

left, right := hello_string.trim_indexes(' f')

assert hello_string.trim(' f') == hello_string.substr(left, right)
```

This is incredibly helpful if you want an indication on what has been trimmed off, it is used inside my project where I need to return correct locations for error reporting on trimmed strings.

\- l-m

### CODE REVIEW

File: vlib/builtin/string.v [hunk 1]
```diff
@@ -1413,11 +1413,21 @@ pub fn (s string) trim_space() string {
...
 		return s.clone()
 	}
+	left, right := s.trim_indexes(cutset)
+	if right - left == 0 {
+		return ''
+	}
+	return s.substr(left, right)
+}
+
+// trim_indices gets the new start and end indicies of a string when any of the characters given in `cutset` were stripped from the start and end of the string. Should be used as an input to `substr()`. If the string contains only the characters in `cutset`, both values returned are zero.
```
Comment (spytheman) [inline vlib/builtin/string.v:1427]:
```suggestion
// trim_indexes gets the new start and end indexes of a string, as if all of the characters given in `cutset` were stripped from the start and end of the string. Should be used as an input to `substr()`. If the string contains only the characters in `cutset`, both values returned are zero.
```

File: vlib/builtin/string.v [hunk 2]
```diff
@@ -1413,11 +1413,21 @@ pub fn (s string) trim_space() string {
...
 // Example: assert ' ffHello V ffff'.trim(' f') == 'Hello V'
-[direct_array_access]
 pub fn (s string) trim(cutset string) string {
 	if s.len < 1 || cutset.len < 1 {
 		return s.clone()
 	}
+	left, right := s.trim_indexes(cutset)
+	if right - left == 0 {
+		return ''
+	}
```
Comment (spytheman) [inline vlib/builtin/string.v:1423]: hm, why is that needed? would not substr work in this case?

Comment (l1mey112) [reply vlib/builtin/string.v:1423]: It should, I just wanted to be more explicit. Either works, your call.

Comment (spytheman) [reply vlib/builtin/string.v:1423]: the less logic, the better imho; if it works without it, please remove it

Comment (l1mey112) [reply vlib/builtin/string.v]: Just made the changes.

**Review by spytheman — APPROVED** [review]
